buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -83,31 +100,71 @@ class BuildPlugin implements Plugin<Project> {
-            if (javaHome == null) {
-                if (System.getProperty("idea.active") != null) {
-                    // intellij doesn't set JAVA_HOME, so we use the jdk gradle was run with
-                    javaHome = Jvm.current().javaHome
-                } else {
-                    throw new GradleException('JAVA_HOME must be set to build Elasticsearch')
-                }
-            }
             project.rootProject.ext.javaHome = javaHome
+            project.rootProject.ext.javaVersion = javaVersion
             project.rootProject.ext.buildChecksDone = true
         }
         project.targetCompatibility = minimumJava
         project.sourceCompatibility = minimumJava
         // set java home for each project, so they dont have to find it in the root project
         project.ext.javaHome = project.rootProject.ext.javaHome
+        project.ext.javaVersion = project.rootProject.ext.javaVersion
     }
 
-    /** Return the name
-     */
-    static String transitiveDepConfigName(String groupId, String artifactId, String version) {
+    /** Finds and enforces JAVA_HOME is set */
+    private static String findJavaHome() {
+        String javaHome = System.getenv('JAVA_HOME')
+        if (javaHome == null) {
+            if (System.getProperty("idea.active") != null) {
+                // intellij doesn't set JAVA_HOME, so we use the jdk gradle was run with
+                javaHome = Jvm.current().javaHome
+            } else {
+                throw new GradleException('JAVA_HOME must be set to build Elasticsearch')
+            }
+        }
+        return javaHome
+    }
+
+    /** Finds printable java version of the given JAVA_HOME */
+    private static String findJavaVersionDetails(Project project, String javaHome) {
+        String versionInfoScript = 'print(' +
+            'java.lang.System.getProperty("java.vendor") + " " + java.lang.System.getProperty("java.version") + ' +
+            '" [" + java.lang.System.getProperty("java.vm.name") + " " + java.lang.System.getProperty("java.vm.version") + "]");'
+        return runJavascript(project, javaHome, versionInfoScript).trim()
+    }
+
+    /** Finds the parsable java specification version */
+    private static String findJavaSpecificationVersion(Project project, String javaHome) {
+        String versionScript = 'print(java.lang.System.getProperty("java.specification.version"));'
+        return runJavascript(project, javaHome, versionScript)
+    }
+
+    /** Finds the parsable java specification version */
+    private static String findJavaVersion(Project project, String javaHome) {
+        String versionScript = 'print(java.lang.System.getProperty("java.version"));'
+        return runJavascript(project, javaHome, versionScript)
+    }
+
+    /** Runs the given javascript using jjs from the jdk, and returns the output */
+    private static String runJavascript(Project project, String javaHome, String script) {
```

Comment:
I think it'd be more normal to create these in build and leave them there? Or not because this all happens during the project setup?